update ZLS install tool

Techatrix opened the pull request:
ZLS builds are now offered by https://github.com/zigtools/release-worker

Vexu reviewed (approved):
Looks good! Have you tested both Windows and Linux?

And do you want me to make a new release of the extension once this is merged?

Techatrix commented:
> Have you tested both Windows and Linux?

Only linux. I will test on Windows and report back soon.

> And do you want me to make a new release of the extension once this is merged?

Yes that would be nice. The only install method doesn't provide new nightly builds and is forever stuck at ZLS 0.13.0

Techatrix commented:
I believe that I have discovered an issue with the windows `tar` utility. Extracting a zip files that is supplied over stdin with `tar -xf - ` will be incorrectly unzipped.

This causes the ZLS installation to only contain the `zls.pdb` file. All other files are missing. The Zig installation is also missing the following files:
- `doc/langref.html`
- `LICENSE`
- `README.md`

This can checked by running the Zig install command in VS-Code and comparing it with 7-Zip or WinRAR. I did not observe this issue when using `bsdtar` on linux.

System Information:
```
OS Name	Microsoft Windows 10 Pro
Version	10.0.19045 Build 19045
```

```
tar --version
# bsdtar 3.5.2 - libarchive 3.5.2 zlib/1.2.5.f-ipp
```

Vexu commented:
I remember running into something similar. Does saving the archive to a file first work?

Techatrix commented:
> Does saving the archive to a file first work?

This works as expected.

Techatrix commented:
I changed to code to temporarily save the tarball/zip on disk until it has been extracted. I tested it on windows and linux.

I also made `downloadAndExtractArtifact` use the async functions instead of blocking and replaced the dependency on `mkdirp` with `vscode.workspace.fs.createDirectory`.
